### PR TITLE
GitHub plugin

### PR DESCRIPTION
**Pull request by kevin-kline**
### Description

This PR introduces a new plugin - The GitHub Plugin. 

### Does this close any open issues?
No. It achieves part of issue #407 

### Current Behavior
No GitHub plugin.

### New Behavior
Collects and saves the following data from GitHub:
- Single Repository
- Commits (for the repository)
- Pull Requests (for the repository)

### Screenshots
image

### Other Information

It uses pagination without the ants package. Ants can be incorporated in future work.

### CODE REVIEW

**Comment by kevin-kline** [comment]
Note that POSTing a request with an object like this will trigger collection:

[{
    "Plugin": "github",
    "Options": {
        "repositoryName": "lake",
        "owner": "merico-dev"
    }
}]

**Comment by kevin-kline** [comment]
This work is in progress.
Collection with ants is working with a fixed rate limit per second. However, we need that to be more dynamic, so I am breaking down the functions to unit test them and fix the implementation of dynamic rate limiting.

**Comment by klesh** [comment]
When would this PR be ready? Can we merge a minimal workable version first, because I also need to work on the issue which certainly requires the api-client code

File: plugins/github/models/github_pull_request.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import "github.com/merico-dev/lake/models"
+
+type GithubPullRequest struct {
+	GithubId        int `gorm:"primaryKey"`
+	RepositoryId    int `gorm:"index"`
+	State           string
+	Title           string
+	HTMLUrl         string
+	MergedAt        string
```
[the remark this answers is not in the document]
Comment (kevin-kline) [reply plugins/github/models/github_pull_request.go:11]: Yes, good idea

**Comment by kevin-kline** [comment]
> When would this PR be ready? Can we merge a minimal workable version first, because I also need to work on the issue which certainly requires the api-client code

Earliest I could finish is probably end of Monday. But if you want to keep working on it, that’s why I pushed it up. So feel free to review and merge, or pull down the branch and continue. I just wanted to be clear that I’m not finished with the rate limiting part, so it may still need some work there. I’ve broken it down so it’s testable to get the rate limit per second from the headers.

**Review by joncodo — APPROVED** [review]

**Pull request merged**